src/components/MapLocation.js

```diff
@@ -109,6 +109,7 @@ function MapLocation() {
         location.transition = null;
       }
       dispatch(setMapLocation(location, resetFormLocation));
+      dispatch(setFormLocation(location));
```

Comment:
I think we should not overwrite the `formLocation` with `mapLocation`. I guess a fix for the bug should be somewhere in `src/store/enhancers/querySync.js`.